src/Compilers/Core/Portable/AssemblyUtilities.cs

```diff
@@ -106,6 +106,7 @@ public static ImmutableArray<string> FindSatelliteAssemblies(string filePath)
             var builder = ImmutableArray.CreateBuilder<string>();
 
             string? directory = Path.GetDirectoryName(filePath);
+            RoslynDebug.AssertNotNull(directory);
```

Comment:
❔ Is it ever possible for `PathUtilities.IsAbsolute(filePath)` to return true but `Path.GetDirectoryName(filePath)` to return null? If not, I almost prefer this form:

```csharp
Debug.Assert(PathUtilities.IsAbsolute(filePath));
string directory = Path.GetDirectoryName(filePath)!;
```

Reply:
If path happens to be `C:\`?